lib/dal/src/attribute/value.rs

```diff
@@ -726,6 +756,74 @@ impl AttributeValue {
         })
     }
 
+    #[instrument(level = "info", skip_all)]
+    async fn get_inferred_input_value(
+        ctx: &DalContext,
+        input_attribute_value_id: AttributeValueId,
+    ) -> AttributeValueResult<Option<Value>> {
+        // let mut maybe_result: Option<Value> = None;
+        let maybe_input_socket_id =
+            match AttributeValue::is_for(ctx, input_attribute_value_id).await? {
+                ValueIsFor::InputSocket(input_socket_id) => Some(input_socket_id),
+                _ => None,
+            };
+
+        let Some(input_socket_id) = maybe_input_socket_id else {
+            return Ok(None);
+        };
+
+        let component_id = Self::component_id(ctx, input_attribute_value_id).await?;
+        let input_socket_match = InputSocketMatch {
+            component_id,
+            input_socket_id,
+            attribute_value_id: input_attribute_value_id,
+        };
+        info!(
+            " AV: Finding inferred input value for input value {}",
+            input_attribute_value_id
+        );
+
+        Ok(
+            match Component::find_inferred_connection_to_input_socket(ctx, input_socket_match).await
+            {
+                Ok(Some(output_match)) => {
+                    info!("AV: found output socket {:?}", output_match);
```

Comment:
We should probably chop down (either in level or get rid of) some of these logging statements